sympy/physics/mechanics/system.py

```diff
@@ -1,8 +1,1009 @@
+    process.
+
+    >>> type(system.eom_method)
+    <class 'sympy.physics.mechanics.kane.KanesMethod'>
+    >>> system.add_holonomic_constraints(
+    ...     bob.masscenter.pos_from(rail.masscenter).dot(system.x)
+    ... )
+    >>> system.q_ind = system.get_joint('pin').coordinates
+    >>> system.q_dep = system.get_joint('slider').coordinates
+    >>> system.u_ind = system.get_joint('pin').speeds
+    >>> system.u_dep = system.get_joint('slider').speeds
+    >>> type(system.eom_method)
+    <class 'NoneType'>
+
+    With the updated system the equations of motion can be formed again.
+
+    >>> system.validate_system()
+    >>> system.form_eoms()
+    Matrix([[-bob_mass*g*l*sin(q_pin) - bob_mass*l**2*u_pin' - bob_mass*l*cos(q_pin)*u_slider' - l*(bob_mass*l*u_pin**2*sin(q_pin) - bob_mass*l*cos(q_pin)*u_pin' - (bob_mass + cart_mass)*u_slider')*cos(q_pin) - l*F*cos(q_pin)]])
+    >>> simplify(system.mass_matrix)
+    Matrix([
+    [bob_mass*l**2*sin(q_pin)**2, -cart_mass*l*cos(q_pin)],
+    [               l*cos(q_pin),                       1]])
+    >>> simplify(system.forcing)
+    Matrix([
+    [-l*(bob_mass*g*sin(q_pin) + bob_mass*l*u_pin**2*sin(2*q_pin)/2 + F*cos(q_pin))],
+    [                                                         l*u_pin**2*sin(q_pin)]])
+
+    """
+
+    def __init__(self, origin=None, frame=None):
+        if origin is None:
+            origin = Point('inertial_origin')
+        elif not isinstance(origin, Point):
+            raise TypeError('Origin must be and instance of Point.')
+        self._origin = origin
+        if frame is None:
+            frame = ReferenceFrame('inertial_frame')
+        elif not isinstance(frame, ReferenceFrame):
+            raise TypeError('Frame must be and instance of ReferenceFrame.')
+        self._frame = frame
+        self._origin.set_vel(self._frame, 0)
+        self._q_ind = Matrix(1, 0, []).T
+        self._q_dep = Matrix(1, 0, []).T
+        self._u_ind = Matrix(1, 0, []).T
+        self._u_dep = Matrix(1, 0, []).T
+        self._kdes = Matrix(1, 0, []).T
+        self._hol_coneqs = Matrix(1, 0, []).T
+        self._nonhol_coneqs = Matrix(1, 0, []).T
+        self._bodies = []
+        self._joints = []
+        self._loads = []
+        self._eom_method = None
+
+    @classmethod
+    def from_newtonian(cls, newtonian):
+        """Constructs the system with respect to a newtonian body."""
+        if isinstance(newtonian, Particle):
+            raise ValueError('A Particle has no frame so cannot act as '
+                             'newtonian.')
+        system = cls(origin=newtonian.masscenter, frame=newtonian.frame)
+        system.add_bodies(newtonian)
+        return system
+
+    @staticmethod
+    def _reset_eom_method(method):
+        """Decorator to reset the eom_method if a property is changed."""
+
+        def wrapper(self, *args, **kwargs):
+            self._eom_method = None
+            return method(self, *args, **kwargs)
+
+        return wrapper
```

Comment:
If this is a decorator should it use `functools.wraps`?

Reply:
That should indeed be added.